apiserver/facades/client/cloud/cloud_test.go

```diff
@@ -1037,206 +1394,270 @@ func (s *cloudSuite) TestRevokeCredentials(c *gc.C) {
+	})
+	expectedLog := []string{` WARNING juju.apiserver.cloud credential cloudcred-meep_julia_three will be deleted but it is used by model deadbeef-0bad-400d-8000-4b1d0d06f00d`}
+	for _, l := range expectedLog {
+		c.Assert(c.GetTestLog(), jc.Contains, l)
 	}
-	s.assertRevokeCredentials(c, t)
 }
 
 func (s *cloudSuite) TestRevokeCredentialsForceMany(c *gc.C) {
-	t := revokeCredentialData{
-		f: func(tag names.CloudCredentialTag) (map[string]string, error) {
-			return map[string]string{
-				coretesting.ModelTag.Id(): "modelName",
-			}, nil
-		},
-		args: []params.RevokeCredentialArg{
-			{Tag: "cloudcred-meep_julia_three", Force: true},
-			{Tag: "cloudcred-meep_bruce_three"},
-		},
-		callsMade: []string{"ControllerTag", "CredentialModels", "RemoveCloudCredential", "RemoveModelsCredential", "CredentialModels"},
-		results: params.ErrorResults{
-			Results: []params.ErrorResult{
-				{},
-				{apiservererrors.ServerError(errors.New("cannot revoke credential cloudcred-meep_bruce_three: it is still used by 1 model"))},
-			},
-		},
-		expectedLog: []string{
-			` WARNING juju.apiserver.cloud credential cloudcred-meep_julia_three will be deleted but it is used by model deadbeef-0bad-400d-8000-4b1d0d06f00d`,
-			` WARNING juju.apiserver.cloud credential cloudcred-meep_bruce_three cannot be deleted as it is used by model deadbeef-0bad-400d-8000-4b1d0d06f00d`,
+	adminTag := names.NewUserTag("admin")
+	defer s.setup(c, adminTag).Finish()
+
+	model := map[string]string{
+		coretesting.ModelTag.Id(): "modelName",
+	}
+
+	tagOne, err := cloudCredentialTag("three", "bruce", "meep", map[string]string{}).CloudCredentialTag()
+	c.Assert(err, jc.ErrorIsNil)
+	tagTwo, err := cloudCredentialTag("three", "julia", "meep", map[string]string{}).CloudCredentialTag()
+	c.Assert(err, jc.ErrorIsNil)
+
+	backend := s.backend.EXPECT()
+	backend.CredentialModels(tagOne).Return(model, nil)
+	backend.CredentialModels(tagTwo).Return(model, nil)
+	backend.RemoveCloudCredential(gomock.Any()).Return(nil)
+	backend.RemoveModelsCredential(gomock.Any()).Return(nil)
+
+	results, err := s.api.RevokeCredentialsCheckModels(params.RevokeCredentialArgs{[]params.RevokeCredentialArg{
+		{Tag: "cloudcred-meep_julia_three", Force: true},
+		{Tag: "cloudcred-meep_bruce_three"},
+	}})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results, gc.DeepEquals, params.ErrorResults{
+		Results: []params.ErrorResult{
+			{},
+			{apiservererrors.ServerError(errors.New("cannot revoke credential cloudcred-meep_bruce_three: it is still used by 1 model"))},
 		},
+	})
+	expectedLog := []string{
+		` WARNING juju.apiserver.cloud credential cloudcred-meep_julia_three will be deleted but it is used by model deadbeef-0bad-400d-8000-4b1d0d06f00d`,
+		` WARNING juju.apiserver.cloud credential cloudcred-meep_bruce_three cannot be deleted as it is used by model deadbeef-0bad-400d-8000-4b1d0d06f00d`,
+	}
+	for _, l := range expectedLog {
```

Comment:
When there's a small and fixed amount, like two, might be worth just having two assert lines, instead of adding a loop structure. 

Up to you though

Reply:
Addressed.